internal/operands/tekton-tasks/tekton-tasks.go

```diff
@@ -199,14 +199,26 @@ func (t *tektonTasks) Cleanup(request *common.Request) ([]common.CleanupResult,
 		objects = append(objects, o)
 	}
 
-	clusterTasks, err := listDeprecatedClusterTasks(request)
-	if err != nil {
-		return nil, err
-	}
-	for _, ct := range clusterTasks {
-		o := ct.DeepCopy()
-		objects = append(objects, o)
+	// Solution to optional Tekton CRD is not implemented yet.
+	// Until then, do not check if Tekton CRD exists.
+	//
+	// if !request.CrdList.CrdExists(tektonCrd) {
```

Comment:
Yes, if crd does not exists, then there are no tasks / clusterTasks/ pipelines

Reply:
> What is the reason why not use this check instead of checking if feature gate is enabled? Does it not work?

We're not watching Tekton CRD, because then it'll throw an error because it thinks it's required CRD and not an optional. Please see https://github.com/kubevirt/ssp-operator/pull/561.

> ClusterTasks should be removed on Cleanup() even if the feature gate is disabled, @ksimon1 is that right?

Yes, but then Tekton CRD is required. I assume that if the customer is using TTO, customer likely will have Tekton CRD already installed so the Feature Gate in ssp-operator will be enabled as well.

> When Tekton CRD does not exist then there are no ClusterTasks to cleanup.

Yes, but we'll need to have `!request.CrdList.CrdExists(tektonCrd)` for it, which doesn't work now because we don't have a solution to optional CRDs.

Reply:
To clarify, `CrdList` interface watches all `CustomResourceDefinition` objects in the cluster. The method `CrdList.CrdExists()` can be used to check if **any** CRD exists in the cluster. 
In this case, we can use it to check if Tekton CRD is present, and only delete `ClusterTasks` if it is.

> We're not watching Tekton CRD, because then it'll throw an error because it thinks it's required CRD and not an optional. Please see https://github.com/kubevirt/ssp-operator/pull/561.

Technically we are not watching objects defined in the Tekton CRD. But we are watching the CRD itself.

Reply:
Yes, you're absolutely right. Let's use `CrdList` interface then.